api/v1alpha1/envoygateway_helpers.go

```diff
@@ -195,6 +212,9 @@ func (r *EnvoyGatewayProvider) GetEnvoyGatewayKubeProvider() *EnvoyGatewayKubern
 
 	if r.Kubernetes == nil {
 		r.Kubernetes = DefaultEnvoyGatewayKubeProvider()
+		if r.Kubernetes.LeaderElection == nil {
```

Comment:
what about the case where `r.Kubernetes` is not nil, but `r.Kubernetes.LeaderElection` is `nil`

Reply:
indeed adding

Reply:
fixed